src/main/java/frc/robot/subsystems/gripper/GripperIOSim.java

```diff
@@ -0,0 +1,62 @@
+package frc.robot.subsystems.gripper;
+
+import static edu.wpi.first.units.Units.Radians;
+import static frc.robot.subsystems.gripper.GripperConstants.*;
+
+import com.ctre.phoenix6.controls.DutyCycleOut;
+import com.ctre.phoenix6.controls.PositionVoltage;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.units.Angle;
+import edu.wpi.first.units.MutableMeasure;
+import edu.wpi.first.units.Units;
+import edu.wpi.first.wpilibj.Timer;
+import lib.motors.TalonFXSim;
+
+public class GripperIOSim implements GripperIO {
+    private final TalonFXSim rollerMotor;
+    private final TalonFXSim angleMotor;
+
+    private final DutyCycleOut powerRequest = new DutyCycleOut(0);
+    private final PositionVoltage positionRequest = new PositionVoltage(0);
+
+    public static PIDController angleController = new PIDController(KP.get(), KI.get(), KD.get());
+
+    public GripperIOSim() {
+        angleMotor =
+                new TalonFXSim(
+                        1,
+                        GripperConstants.ANGLE_MOTOR_GEAR_RATIO,
+                        0.5,
+                        GripperConstants.ANGLE_MOTOR_GEAR_RATIO);
+
+        rollerMotor = new TalonFXSim(1, 1, 0.5, 1);
+
+        angleMotor.setController(angleController);
+    }
+
+    @Override
+    public void setRollerMotorPower(double power) {
+        powerRequest.withOutput(power);
+        rollerMotor.setControl(powerRequest);
+    }
+
+    @Override
+    public void setAngleMotorPower(double power) {
+        powerRequest.withOutput(power);
+        angleMotor.setControl(powerRequest);
+    }
+
+    @Override
+    public void setAngle(MutableMeasure<Angle> angle) {
+        positionRequest.withPosition(angle.in(Radians));
+        positionRequest.withPosition(Math.IEEEremainder(angle.in(Units.Degrees), 2 * Math.PI));
```

Comment:
Why twice?

Reply:
855bf1d768d6b98af4dcefe8fbafa400ab70426a

Reply:
755fe13602396b0d2ef3c27f97afed1412bb95e0

Reply:
Not a relevant commit...

Reply:
yes it is?

Reply:
We decided to work with radians this year and I don't mind that

Reply:
It's much more intuitive to work in degrees... if we're in the competition and need to change some setpoints do you really think it'll be convenient to work in radians? Probably what'll happen is that you'll work in degrees and convert them to radians.

Reply:
ae0bf35af6aa89681864609b8e4f301892bbfe18

Reply:
```diff
diff --git a/src/main/java/frc/robot/subsystems/gripper/GripperIOSim.java b/src/main/java/frc/robot/subsystems/gripper/GripperIOSim.java
index 56451be..ce33294 100644
--- a/src/main/java/frc/robot/subsystems/gripper/GripperIOSim.java
+++ b/src/main/java/frc/robot/subsystems/gripper/GripperIOSim.java
@@ -50,14 +50,13 @@ public class GripperIOSim implements GripperIO {
     @Override
     public void updateInputs() {
         rollerMotor.update(Timer.getFPGATimestamp());
         angleMotor.update(Timer.getFPGATimestamp());
-        inputs.currentAngle.mut_replace(angleMotor.getPosition(), Degrees);
+        inputs.currentAngle.mut_replace(angleMotor.getPosition(), Radians);
         inputs.rollerMotorVoltage.mut_replace(rollerMotor.getAppliedVoltage(), Volts);
         inputs.angleMotorVoltage.mut_replace(angleMotor.getAppliedVoltage(), Volts);
     }
```
If I understand correctly, `angleMotor.getPosition()` returns a unitless measure, so I don't think simply switching the unit from degrees to radians will work the way you want it to. Did you check it in the simulation?
However, I'd want @vichik123's input on this too (degrees/radians). Generally, my intent is never to force you to make changes, but for us to reach a mutual understanding on what is the best solution.